### PR TITLE
feat(rbac): add rbac dynamic plugin

### PR DESCRIPTION
**Pull request by ciiay**
## Description

By this change, the RBAC dynamic plugin is included in the backstage-showcase container image, disabled by default.

## Which issue(s) does this PR fix

- This is for [RHIDP-1050](https://issues.redhat.com/browse/RHIDP-1050): Add/enable RBAC UI plugin to showcase/RHDH and test the RBAC UI flow

## Requires
https://github.com/janus-idp/backstage-plugins/pull/1133

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [x] Documentation is updated if necessary (requirement for new features)
- [x] Add a screenshot if the change is UX/UI related

## How to test changes / Special notes to the reviewer
1. Enable authentication and login with a non-guest user.
2. In your local `backstage-plugins/plugins/rbac` directory, run `yarn export-dynamic` to export RBAC plugin UI as a dynamic plugin
3. In your local `backstage-showcase/dynamic-plugins-root/` directory, run the below command to install RBAC plugin UI
```
pkg=<local/path/to/backstage-plugins/plugins/rbac/>
archive=$(npm pack $pkg)
tar -xzf "$archive" && rm "$archive"
mv package $(echo $archive | sed -e 's:\.tgz$::')
```
4. In your local `backstage-showcase/app-config.local.yaml` file, add the below changes
```
...
# Add this to enable RBAC
permission:
  enabled: true
  rbac:
    policies-csv-file: ../../rbac-policy.csv
    # In ../../rbac-policy.csv add the following line to test RBAC backend installation:
    # p, role:default/team_a, catalog-entity, read, allow

    # g, user:default/<USERNAME>, role:default/team_a

    # p, role:default/team_a, catalog-entity, read, allow
    # p, role:default/team_a, catalog.entity.create, create, allow
    # p, role:default/team_a, catalog.location.create, create, allow
    # p, role:default/team_a, catalog.location.read, read, allow
    admin:
      users:
        - name: user:default/ciiay
# End of RBAC config

# Add this to enable Dynamic Plugins
dynamicPlugins:
  rootDirectory: dynamic-plugins-root
  frontend:
    janus-idp.backstage-plugin-rbac:
      appIcons:
        - name: AdminPanelSettingsOutlinedIcon
          module: RbacPlugin
          importName: AdminPanelSettingsOutlinedIcon
      dynamicRoutes:
        - path: /rbac
          importName: RbacPage
          module: RbacPlugin
          menuItem:
            icon: AdminPanelSettingsOutlinedIcon
            text: Administration
# End of Dynamic Plugins enablement config
```
5. In `dynamic-plugins.default.yaml` file Line 503, change disabled value to `disabled: false` to enable RBAC plugin UI.
6. Start Backend-showcase app, and observe that `Administration` button shows on the left side menu.

**Screen recording of test:**

https://github.com/janus-idp/backstage-showcase/assets/26255262/c0abc670-b429-4073-b5a0-b65253d15f91

### CODE REVIEW

**Comment by yuvalgalanti** [comment]
A few questions I have:

1. why the row height in the rolls list/table is so big can we reduce that? 
2. I think that the administration section should be located above the "settings" area

**Comment by parvathyvr** [comment]
Agree with @yuvalgalanti  on point 2- as per our design and earlier implementations i have seen- Administration is placed above Settings in the left navigation.

**Comment by gashcrumb** [comment]
Just confirming that once https://github.com/janus-idp/backstage-showcase/pull/932 is available we can rework how the tab is configured for the rbac plugin to move it where it needs to be.  If this is merged sooner I can probably do it as part of https://github.com/janus-idp/backstage-showcase/issues/941

File: dynamic-plugins/imports/package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
     "@janus-idp/backstage-plugin-ocm": "3.6.1",
     "@janus-idp/backstage-plugin-ocm-backend": "3.4.12",
     "@janus-idp/backstage-plugin-quay": "1.5.2",
+    "@janus-idp/backstage-plugin-rbac": "1.11.1",
```
Comment (gashcrumb) [inline dynamic-plugins/imports/package.json:29]: I think this should be 1.13.0.

**Review by gashcrumb — CHANGES_REQUESTED** [review]
I think the version needs to be bumped, otherwise it's looking good!  I'll do a bit more testing with a local change just to verify that.

File: dynamic-plugins/imports/package.json (hunk 1 above)
Comment (ciiay) [reply dynamic-plugins/imports/package.json]: @gashcrumb Thanks for the review. Good catch. I have bumped up the version and tested locally. PR updated.

**Review by gashcrumb — APPROVED** [review]
have it working locally now too, looks great!

**Review by gashcrumb — APPROVED** [review]

**Comment by debsmita1** [comment]
Was able to test it locally. Thanks @ciiay 

Screenshot 2024-02-05 at 3 04 34 PM

/lgtm

**Pull request merged**